lib/grandpa/message_handler.go

```diff
@@ -246,13 +246,35 @@ func (h *MessageHandler) verifyCatchUpResponseCompletability(prevote, precommit
 	return nil
 }
 
+func getEquivocatoryVoters(votes []AuthData) map[ed25519.PublicKeyBytes]struct{} {
+	eqvVoters := make(map[ed25519.PublicKeyBytes]struct{})
+	voters := make(map[ed25519.PublicKeyBytes]int)
```

Comment:
nit that one can be pre-allocated:

```suggestion
	voters := make(map[ed25519.PublicKeyBytes]int, len(votes))
```